src/test/java/uk/co/ramp/OutbreakTest.java

```diff
@@ -97,9 +97,17 @@ public void runContactDataWithRandoms() {
             .filter(status -> status == SUSCEPTIBLE)
             .count();
 
-    // we expect this to roughly follow an exp decay
-    double test = popSize * Math.exp(-randomInfection * days);
-    Assert.assertEquals(test / (double) popSize, sus / (double) popSize, 0.1);
+    // We expect this to roughly follow an exp decay, but with a large tolerance.
+    //
+    // If r='randomInfection' ratio of people become other status than SUSCEPTIBLE,
+    // after T days, remaining population of SUSCEPTIBLE persons should be
+    // (1-r)^T = exp(T log(1-r))
+    // Previously, an approximation log(1-r)=r was applied but this is valid only when r is small
+    double test = popSize * Math.exp(Math.log(1. - randomInfection) * days);
+
+    // Large tolerance 0.5 should be allowed given the mechanism of contact-based exposure, in
+    // addition to the random infection
+    Assert.assertEquals(test / (double) popSize, sus / (double) popSize, 0.5);
```

Comment:
If we run this for larger sample size, can we reduce the tolerance? 0.5 is very large. Always better to have a narrow tolerance in tests so it is easy to detect when something breaks. What are your thoughts on this? Happy to discuss further.

Reply:
Yes, 0.5 is large but under the new model 0.1 cannot be achieved because there are new infections by both random without-contact one and contact-based person-to-person one. When the impact of the latter gets stronger, exponential decay approximation is very erroneous.

One alternative considerable is to calibrate the logarithmic decay rate and set the tolerance on
the difference between `log(test / popSize) / n_days` and `randomInfection`. Another alternative is to calibrate the difference in early period. If early-stage exposure involves less second-order effect than the late-stage one, so exponential decay should have a higher approximation accuracy. Instead of taking the 11 days, we can try to pick up the first 3 or 4 days and calibrate the decay rate in logarithmic sense. Detection of crucially wrong modification should be possible even in this simplified version.